cmd/cli/root.go

```diff
@@ -320,6 +320,11 @@ func (a *App) assign(c *cli.Context) error {
 		return fmt.Errorf("%q: invalid repository name", a.Config.Downstream.GitHubRepoName)
 	}
 
+	upstreamRepoName, err := gh.ParseURL(a.Config.Upstream.URL)
```

Comment:
I think we should add a new setting for that instead of trying to parse from the URL.

Reply:
You mean changing the upstream URL to a an object (as we do for downstream)? This will require changing multiple places in the code.